internal/controller/warehouses/git.go

```diff
@@ -26,8 +26,17 @@ func (r *reconciler) selectCommits(
 	ctx context.Context,
 	namespace string,
 	subs []kargoapi.RepoSubscription,
+	lastFreight *kargoapi.FreightReference,
 ) ([]kargoapi.GitCommit, error) {
 	latestCommits := make([]kargoapi.GitCommit, 0, len(subs))
+
+	repoCommitMappings := map[string]string{}
+	if lastFreight != nil {
+		for _, commit := range lastFreight.Commits {
+			repoCommitMappings[commit.RepoURL+"#"+commit.Branch] = commit.ID
```

Comment:
Nit: as the size of the map always equals the number of commits, you could do:

```suggestion
	var repoCommitMappings map[string]string
	if lastFreight != nil {
		repoCommitMappings = make(repoCommitMappings, len(lastFreight.Commits))
		for _, commit := range lastFreight.Commits {
			repoCommitMappings[commit.RepoURL+"#"+commit.Branch] = commit.ID
```